src/Basic/Validation.cs

```diff
@@ -0,0 +1,24 @@
+using AslHelp.Data.AutoSplitter;
+
+namespace AslHelp;
+
+public static class Validation
+{
+    public static void AssertNotNull(string name, dynamic value)
```

Comment:
call this `NotNull`. use `object` instead of `dynamic`. use `CallerArgumentExpressionAttribute` for automatic argument name inference.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callerargumentexpressionattribute?view=net-8.0 appears to be in a future version of .net, am I wrong on this?

Reply:
annoying. polysharp fixes this, but introducing that to asl-help would open up way too many things that i won't deal with (nullability). i already use it extensively in v2, but won't bother with it here.